docs/pages/image-extractor.rst

```diff
@@ -145,23 +147,94 @@ and DataLoader, refer to this guide: https://pytorch.org/tutorials/beginner/data
 
     for i, sample_batch in enumerate(dataloader):
         img, label = sample_batch["rgba"], sample_batch["label"]
-        img = img.permute(0, 3, 1, 2)  # Reshape to PyTorch format for convolutions
+        img = img.permute(0, 3, 1, 2)
         out = net(img)
 
 
+`Writing Custom Pose Extractors`_
+=================================
+
+The pose extractor defines how camera poses are programmatically determined so that the image
+extractor knows how to manipulated the camera position and angle to extract an image from the
+simulator. All custom pose extractors must inherit from the PoseExtractor abstract class and
+override the extract_poses method. Further, you must register the pose extractor using
+habitat_sim.registry.
+
+The extract_poses method takes three required arguments:
+
+* labels: List[float]: A list of numeric labels of points we are looking for
+* view: numpy.ndarray: A 2 dimensional array representing the topdown view of a scene.
+* fp: str: The filepath of the scene.
+
+The job of the extract_poses method is to return a list of poses where each pose is a four-tuple
+of (<camera position>, <point of interest>, <label>, <scene filepath>). The camera position is the
+coordinates of the camera in the space of the topdown view. The point of interest is the position where
```

Comment:
Is there a reason we're working in 2D coordinates (as opposed to 6DoF camera pose)? What if someone wants to change the height of the camera or tilt?

Reply:
Yes, the topdown map that defines which points in the scene are navigable and unnavigable is how we determine which camera positions are valid. When I was developing the image extractor, I saw no simple way of determining valid height information. As for adjusting the tilt, that would require a change to the _convert_to_scene_coordinate_system method in PoseExtractor, specifically in how we compute the camera angle, which is entirely possible but getting the camera angle is non-trivial so I opted for a simpler solution during my internship. I think this is a great place for future improvement however!

Reply:
> Yes, the topdown map that defines which points in the scene are navigable and unnavigable is how we determine which camera positions are valid. 

Right but philosophically, camera placements do not need to be restricted to positions where the agent can "stand". 

> As for adjusting the tilt, that would require a change to the _convert_to_scene_coordinate_system method in PoseExtractor, specifically in how we compute the camera angle, which is entirely possible but getting the camera angle is non-trivial so I opted for a simpler solution during my internship. I think this is a great place for future improvement however!

It's a perfectly fine first step. Can we just state in the sentences how someone might go about changing this if they want to?